### PR TITLE
Add ReadWorkspaces and ReadProjects to Orgs

### PR DESCRIPTION
**Pull request by juliannatetreault**
## Description
Adds `ReadWorkspaces` and `ReadProjects` permissions to Organizations in `team.go` and updates the accompanying test, `team_integration_test.go`. These are all part of a larger change which gives read access at an organization level.

This change is in draft as we complete the feature and prepare for its release.

## Testing plan
1.  Verify that you can set the properties by using `go-tfe` within the TFE provider (caveat: this isn't doable yet!)
2. Verify that all tests still pass.

## External links
- [RFC](https://go.hashi.co/rfc/tf-596)
- [Jira](https://hashicorp.atlassian.net/browse/TF-1440)

## Output from tests
Including output from tests may require access to a TFE instance. Ignore this section if you have no environment to test against.
```
=== RUN   TestTeam_Unmarshal
--- PASS: TestTeam_Unmarshal (0.00s)
PASS

Process finished with the exit code 0

=== RUN   TestTeamCreateOptions_Marshal
--- PASS: TestTeamCreateOptions_Marshal (0.00s)
PASS

Process finished with the exit code 0
```

### CODE REVIEW

**Review by nfagerlund — CHANGES_REQUESTED** [review]
This looks good! I'm marking as request changes because of the time-traveling changelog line (which now that I think on it is likely just due to the merge from upstream). Once that's sorted out, I'll be 👍🏼 to merge!

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@
 ## Enhancements
 * Adds `List()` method to `GPGKeys` interface by @sebasslash [#602](https://github.com/hashicorp/go-tfe/pull/602)
 * Adds `ProviderBinaryUploaded` field to `RegistryPlatforms` struct by @sebasslash [#602](https://github.com/hashicorp/go-tfe/pull/602)
+* Adds `ReadWorkspaces` and `ReadProjects` permissions to `Organizations` by @JuliannaTetreault [#614](https://github.com/hashicorp/go-tfe/pull/614)
```
[the remark this answers is not in the document]
Comment (juliannatetreault) [reply CHANGELOG.md]: Ah, whoops! I've made this change, @nfagerlund. 😄

File: team.go [hunk 2]
```diff
@@ -70,6 +70,8 @@ type OrganizationAccess struct {
 	ManageRunTasks        bool `jsonapi:"attr,manage-run-tasks"`
 	// **Note: This field is still in BETA and subject to change.**
 	ManageProjects bool `jsonapi:"attr,manage-projects"`
+	ReadWorkspaces        bool `jsonapi:"attr,read-workspaces"`
```
Comment (nfagerlund) [inline team.go:73]: This needs a gofmt run! merging from upstream is always an adventure, huh

Comment (juliannatetreault) [reply team.go]: Ah, sorry about that! TIL!

**Comment by juliannatetreault** [comment]
> Well, this WOULD be good to go, except that our CI is broken at the moment. Team's looking into it and I'll kick the jobs when it's working again.

No worries at all! Thank you so much for the update and reviews 🌟

**Pull request merged**